### PR TITLE
feat(organization): general settings

### PR DESCRIPTION
**Pull request by RemiBonnet**
# What does this PR do?

[> Link to the JIRA ticket
](https://qovery.atlassian.net/jira/software/projects/FRT/boards/23?selectedIssue=FRT-491)

- Add input-file component
- Add input-tags component
- Add section General organization setting

Capture d’écran 2022-09-22 à 11 14 35

---

### Global

- [x] This PR does not introduce any breaking change
- [ ] This PR introduces breaking change(s) and has been labeled as such
- [x] I have found someone to review this PR and pinged him

### Store

- [ ] This PR introduces new store changes

### NX

- [x] I have run the dep-graph locally and made sure the tree was clean i.e no circular dependencies
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`

### Clean Code

- [x] I made sure the code is type safe (no any)
- [ ] I have included a feature flag on my feature, if applicable

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [[PR] staging - feat(organization): general settings - 2022-09-15T15:59:58Z](https://console.qovery.com/platform/organization/141c07c8-0dd9-4623-983b-3fdd61867255/projects/977bd069-20f7-4b24-bd95-28b44174cf43/environments/b8b354ce-1303-4179-99dd-d3e1bd821663/applications)

 Another comment will be posted when deployments are terminated

**Comment by evoxmusic** [comment]
Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://zb8b354ce-z15d1340a-gtw.bool.sh)
:point_right: [storybook](https://zb8b354ce-ze2714b9f-gtw.bool.sh)

File: libs/domains/organization/src/lib/slices/organization.slice.ts [hunk 1]
```diff
@@ -48,6 +51,19 @@ export const fetchContainerRegistries = createAsyncThunk<ContainerRegistryRespon
   }
 )
 
+export const editOrganization = createAsyncThunk(
+  'organization/edit',
+  async (payload: { organizationId: string; data: Partial<OrganizationEntity> }) => {
+    const cloneOrganization = Object.assign({}, refactoOrganizationPayload(payload.data))
```
Comment (bdebon) [inline libs/domains/organization/src/lib/slices/organization.slice.ts:57]: cool to have this util for all the edit

File: libs/pages/environments/src/lib/ui/create-clone-environment-modal/create-clone-environment-modal.spec.tsx [hunk 2]
```diff
@@ -48,14 +48,15 @@ describe('CreateCloneEnvironmentModal', () => {
   })
 
   it('should reformat name by replacing special char by hyphens', async () => {
-    const { baseElement } = render(
+    const { baseElement, debug } = render(
```
Comment (bdebon) [inline libs/pages/environments/src/lib/ui/create-clone-environment-modal/create-clone-environment-modal.spec.tsx:51]: You can remove the debug

File: libs/pages/environments/src/lib/ui/create-clone-environment-modal/create-clone-environment-modal.spec.tsx [hunk 3]
```diff
@@ -48,14 +48,15 @@ describe('CreateCloneEnvironmentModal', () => {
...
       wrapWithReactHookForm(<CreateCloneEnvironmentModal {...props} />, {
         defaultValues,
       })
     )
     const input = getByTestId(baseElement, 'input-text')
     await act(() => {
       fireEvent.input(input, { target: { value: 'ben et remi' } })
+      debug()
```
Comment (bdebon) [inline libs/pages/environments/src/lib/ui/create-clone-environment-modal/create-clone-environment-modal.spec.tsx:59]: idem

File: libs/pages/settings/src/lib/feature/page-organization-general-feature/page-organization-general-feature.tsx [hunk 4]
```diff
@@ -1,7 +1,77 @@
...
+import { useEffect, useState } from 'react'
+import { FieldValues, FormProvider, useForm } from 'react-hook-form'
+import { useDispatch, useSelector } from 'react-redux'
+import { useParams } from 'react-router-dom'
+import { editOrganization, selectOrganizationById } from '@qovery/domains/organization'
+import { useDocumentTitle } from '@qovery/shared/utils'
+import { AppDispatch, RootState } from '@qovery/store/data'
 import PageOrganizationGeneral from '../../ui/page-organization-general/page-organization-general'
 
+export const handleSubmit = (data: FieldValues, organization: Organization) => {
```
Comment (bdebon) [inline libs/pages/settings/src/lib/feature/page-organization-general-feature/page-organization-general-feature.tsx:11]: why you don't use the spread operator function here?
```
return {
   ...organization,
   logoUrl: data.logo_url,
   name: data.name,
   etc
}
```

It does the same thing but just quicker to write and read

**Review by bdebon — APPROVED** [review]
Niiiice! Super slick interface and good effort put in this new tag component! 
Welle done with the tests too. Good good good

**Comment by bdebon** [comment]
I saw a url pattern on this input but strangely it does not show an error here:
CleanShot 2022-09-22 at 15 21 27@2x

**Pull request merged**